src/ripple/protocol/impl/STPathSet.cpp

```diff
@@ -156,10 +156,10 @@ STPath::getJson (int) const
     for (auto it: mPath)
     {
         Json::Value elem (Json::objectValue);
-        int         iType   = it.getNodeType ();
+        const auto iType   = it.getNodeType ();
 
         elem[jss::type]      = iType;
-        elem[jss::type_hex]  = strHex (iType);
+        elem[jss::type_hex]  = charHex (iType);
```

Comment:
I think this is a bug. This will only values < 16 to a single hex character.

Reply:
Good catch. Thanks for the fix.